--- a/e3sm/configure.ac
+++ b/e3sm/configure.ac
@@ -795,7 +795,7 @@ if test "x$have_hdf5" = xyes ; then
       if test "x$extra_AM_ldflags" != x ; then
          extra_ldflags+=$extra_AM_ldflags
       fi
-      extra_libs="`grep 'Extra libraries' $hdf5_lib/libhdf5.settings |cut -f2 -d:`"
+      extra_libs="`grep 'Extra libraries' $hdf5_lib/libhdf5.settings |cut -f2 -d: | sed -e 's/;/ /g' -e 's#\\(^\\| \\)m\\($\\| \\)# -lm #g' -e 's#\\(^\\| \\)dl\\($\\| \\)# -ldl #g'`"
    fi
    if test "x$LIBS" = x ; then
       LIBS="-lhdf5 $extra_libs"
